packages/embark/src/lib/modules/code_generator/index.js

```diff
@@ -346,20 +350,46 @@ class CodeGenerator {
         code += self.generateStorageInitialization(true);
         code += self.generateNamesInitialization(true);
         code += self.getReloadPageCode();
+        code += "\nexport default EmbarkJS;";
+        code += "\nif (typeof module !== 'undefined' && module.exports) {" +
+          "\n\tmodule.exports = EmbarkJS;" +
+          "\n}";
         code += '\n/* eslint-enable */';
 
         next();
       },
       function writeFile(next) {
         self.generateArtifact(code, constants.dappArtifacts.embarkjs, '', next);
+      },
+      function transformCode(artifactPath, updated, next) {
+        if (!updated) {
+          return next();
+        }
+        transform(code, {
+          cwd: self.fs.embarkPath(),
+          "presets": [
+            [
+              "@babel/preset-env", {
+              "targets": {
+                "node": "6.10"
```

Comment:
One thing I just noticed is that `"6.10"` is a bit mismatched with our common babel config: https://github.com/embark-framework/embark/blob/master/babel.config.js#L55.

Also, at runtime embark enforces the minimum node version to be `8.11.3`. So maybe this target should match?

Reply:
Done. I also squashed my commits so it's ready to merge